app/src/androidTest/java/com/github/palFinderTeam/palfinder/profile/UIMockProfileServiceModule.kt

```diff
@@ -127,8 +127,15 @@ object UIMockProfileServiceModule {
                 if (!user.canFollow(targetId)) {
                     return Response.Failure("Cannot follow this user.")
                 }
+                val targetProfile = fetch(targetId)!!
                 db[user.uuid] = user.copy(following = user.following.plus(targetId))
+                if (updateAchievementsFollower(user) != null) {
+                    db[user.uuid] = user.copy(achievements = user.achievements().map{it.string}.plus(updateAchievementsFollower(user)!!))
```

Comment:
Not very clear, is it possible to rewrite this in a more understandle format?

Reply:
the thing is that the achievement system is not thought for the UI mock repository, which is only meant for testing purpose. So I had to adapt this function to reach the same value with some mapping